pkg/apis/servicecatalog/validation/broker_test.go

```diff
@@ -32,6 +32,8 @@ func TestValidateBroker(t *testing.T) {
 		valid  bool
 	}{
 		{
+			// covers the case where there is no AuthInfo field specified. the validator should
+			// ignore the field and still succeed the validation
 			name: "valid broker - no auth secret",
```

Comment:
not a feedback on your PR, but I have had a lot of luck with a method that returns a valid resource, and then subtracting / mutating things to create objects for invalid scenarios.